### ISSUE TITLE
Support pull image

### ISSUE THREAD

**Issue by pilgrim2go** [aws/amazon-ecs-agent#804]
Is it possible if we 
register new Task Definition ( or new revision from existing Task)
and pull image only?

It's same as docker-compose pull app. we need it to save time to pull large image ( GB)  over AWS regions.

**Comment by jhaynes** [comment]
Hi @pilgrim2go. We do not currently have a way to pre-cache images on a host. However, the pull timeout is quite long. 

Can you explain more about your specific use case? What issue are you running into that you want to pre-cache images?

**Comment by pilgrim2go** [comment]
hi @jhaynes : We have multi deployments in multi regions. And our images are hosted in ECS Singapore ( ap-southeast-1).
When deploying in EU, it takes to pull image in Singapore ( 1.5GB) to EU though it doesn't take a long time. Just wonder if we can have something like pre-cache so at the time we deploy app, the process takes less time.

**Comment by samuelkarp** [comment]
Thanks, I've marked this as a feature request.

**Comment by aaronwalker** [comment]
@samuelkarp The use case we have for this would be tagging a service to cache it's images on the ecs cluster instances to speed up autoscaling to avoid having to download the image on that instance. Or even bake AMIs with the images already cached. We often have to deal with quite large containers that aren't deployed to ECR. I'd be happy to have ago at implementing support for this if.

**Comment by miklis** [comment]
+1 for this. We also have some large images used in an autoscaling ECS cluster. Whenever a new instance is registered to the cluster, it would speed up the first tasks slightly if we would be able to specify to start pulling some images already at startup.

**Comment by yunluyl** [comment]
+1 We have a 5GB docker image. Without pre-caching, the first container bringup on a new instance takes 3+ minutes.

**Comment by coultn** [comment]
It has been a while since this issue was opened - everyone has my sincere thanks for the feedback.   We on the ECS team are aware of this request and are actively considering solutions to the issue.  We always appreciate +1's and additional details/use cases - they will help inform our work moving forward.

**Comment by CpuID** [comment]
I agree this is a useful feature to have to speed up recoveries/scale up events, I do think when implementing it consideration needs to be given to which/how many images to cache for disk space reasons.

Eg. if you have 100 apps, and you tried to cache all apps on all hosts, you may end up running out of local disk to store them.

Some way to flag if an image should be cached at the "edge" (eg. the EC2 hosts) potentially in the ECS task definition seems like the right answer potentially...? With a default off/false?

I'm sure this would make it far more tricky to implement on Fargate also (due to the shared nature of the underlying host pools).

**Comment by juultje123** [comment]
Is there any news on this issue? It would be very helpful to have pre-cached containers. But indeed not easy to implement, since there is no single host underneath Fargate for holding these cached containers.

**Comment by yaup** [comment]
+1 we're using EC2 and have really large images that would benefit from pre-caching. We're hacking around it by pre-caching it ourselves with a cron job, but getting it to setup aws credentials to pull from ECR all in the launch configuration user data is not ideal and really hacky.

**Comment by amunhoz** [comment]
+1 really need that.

**Comment by petderek** [comment]
> We're hacking around it by pre-caching it ourselves with a cron job, but getting it to setup aws credentials to pull from ECR all in the launch configuration user data is not ideal and really hacky.

Another hacky way you can do this is by registering a secondary task definition that pulls your container, but doesn't run the application (ie, use a different command like echo). This would at least mean fewer hacks around auth.
